src/main/kotlin/com/mparticle/kits/IterableKit.kt

```diff
@@ -155,21 +155,39 @@ class IterableKit : KitIntegration(), ActivityListener, ApplicationStateListener
                     id = MParticle.getInstance()?.Identity()?.deviceApplicationStamp
                 }
             }
-            if (id != null) {
-                "$id@placeholder.email"
-            } else {
-                null
-            }
+            id
+        }
+    }
+
+    private fun getPlaceholderEmail(userId: String?): String? {
+        if (userId != null) {
+            return "$userId@placeholder.email"
+        } else {
+            return null
         }
     }
 
     private fun updateIdentity(mParticleUser: MParticleUser) {
-        val userIdentities = mParticleUser.userIdentities
-        val email = userIdentities[IdentityType.Email]
-        val placeholderEmailFt = getPlaceholderEmail(mParticleUser)
-        placeholderEmailFt
-            .onSuccess(object : Future.SuccessCallback<String?> {
-                override fun onSuccess(placeholderEmail: String?) {
+        val userId = getUserId(mParticleUser)
+
+        if (prefersUserId) {
```

Comment:
Could move the if condition to the onSuccess block, and have only one onSuccess/onFailure instead of two.